aws/resource_aws_neptune_cluster_instance.go

```diff
@@ -0,0 +1,560 @@
+			Timeout:    d.Timeout(schema.TimeoutUpdate),
+			MinTimeout: 10 * time.Second,
+			Delay:      30 * time.Second,
+		}
+
+		// Wait, catching any errors
+		_, err = stateConf.WaitForState()
+		if err != nil {
+			return err
+		}
+
+	}
+
+	if err := setTagsNeptune(conn, d, d.Get("arn").(string)); err != nil {
+		return err
+	}
+
+	return resourceAwsNeptuneClusterInstanceRead(d, meta)
+}
+
+func resourceAwsNeptuneClusterInstanceDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).neptuneconn
+
+	log.Printf("[DEBUG] Neptune Cluster Instance destroy: %v", d.Id())
+
+	opts := neptune.DeleteDBInstanceInput{DBInstanceIdentifier: aws.String(d.Id())}
+
+	log.Printf("[DEBUG] Neptune Cluster Instance destroy configuration: %s", opts)
+	if _, err := conn.DeleteDBInstance(&opts); err != nil {
+		return err
+	}
+
+	log.Println("[INFO] Waiting for Neptune Cluster Instance to be destroyed")
+	stateConf := &resource.StateChangeConf{
+		Pending:    resourceAwsNeptuneClusterInstanceDeletePendingStates,
+		Target:     []string{},
+		Refresh:    resourceAwsNeptuneInstanceStateRefreshFunc(d.Id(), conn),
+		Timeout:    d.Timeout(schema.TimeoutDelete),
+		MinTimeout: 10 * time.Second,
+		Delay:      30 * time.Second,
+	}
+
+	if _, err := stateConf.WaitForState(); err != nil {
+		return err
+	}
+
+	return nil
+
+}
+
+var resourceAwsNeptuneClusterInstanceCreateUpdatePendingStates = []string{
+	"backing-up",
+	"configuring-enhanced-monitoring",
+	"creating",
+	"maintenance",
+	"modifying",
+	"rebooting",
+	"renaming",
+	"starting",
+	"upgrading",
+}
+
+var resourceAwsNeptuneClusterInstanceDeletePendingStates = []string{
+	"modifying",
+	"deleting",
+}
+
+func resourceAwsNeptuneInstanceStateRefreshFunc(id string, conn *neptune.Neptune) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		v, err := resourceAwsNeptuneInstanceRetrieve(id, conn)
+
+		if err != nil {
+			log.Printf("Error on retrieving Neptune Instance when waiting: %s", err)
+			return nil, "", err
+		}
+
+		if v == nil {
+			return nil, "", nil
+		}
+
+		if v.DBInstanceStatus != nil {
+			log.Printf("[DEBUG] Neptune Instance status for instance %s: %s", id, aws.StringValue(v.DBInstanceStatus))
+		}
+
+		return v, aws.StringValue(v.DBInstanceStatus), nil
+	}
+}
+
+func resourceAwsNeptuneInstanceRetrieve(id string, conn *neptune.Neptune) (*neptune.DBInstance, error) {
+	opts := neptune.DescribeDBInstancesInput{
+		DBInstanceIdentifier: aws.String(id),
+	}
+
+	log.Printf("[DEBUG] Neptune Instance describe configuration: %#v", opts)
+
+	resp, err := conn.DescribeDBInstances(&opts)
+	if err != nil {
+		if isAWSErr(err, neptune.ErrCodeDBInstanceNotFoundFault, "") {
+			return nil, nil
+		}
+		return nil, fmt.Errorf("Error retrieving Neptune Instances: %s", err)
+	}
+
+	if len(resp.DBInstances) != 1 ||
+		aws.StringValue(resp.DBInstances[0].DBInstanceIdentifier) != id {
+		if err != nil {
```

Comment:
Anytime `err != nil`, it is previously `return`ed -- this conditional seems like it should be removed.